src/Components/ContextProvider/ContextProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+  FC,
+  ReactNode,
+  useState,
+  useEffect,
+  useMemo,
+} from 'react';
+
+import {
+  getTodos,
+  updateTodo,
+} from '../../api/todos';
+
+import { Context } from '../../Context';
+
+import { Todo } from '../../types/Todo';
+import { ErrorMessage } from '../../types/ErrorMessage';
+import { Filter } from '../../types/Filter';
+
+interface Props {
+  children: ReactNode;
+}
+
+export const ContextProvider: FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<Filter>(Filter.ALL);
+  const [
+    errorMessage,
+    setErrorMessage,
+  ] = useState('');
+
+  const USER_ID = 12176;
+
+  const handleErrorChange = (value: string) => {
+    setErrorMessage(value);
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => handleErrorChange(ErrorMessage.UNABLE_TO_LOAD));
+  }, [todos]);
+
+  const handleActiveTodos = useMemo(() => {
+    return todos.reduce((sum, item) => {
+      if (!item.completed) {
+        return sum + 1;
+      }
+
+      return sum;
+    }, 0);
+  }, [todos]);
+
+  const completeAll = () => {
```

Comment:
It shouldn't be here - it's related only to `Header` (used only there), so define this function in the Header component